server/sonar-server/src/main/java/org/sonar/server/rule/RuleOperations.java

```diff
@@ -111,7 +111,7 @@ public boolean updateRule(RuleDto ruleDto, @Nullable CharacteristicDto newSubCha
       // No sub-characteristic is given -> disable rule debt if not already disabled
     } else {
       // Rule characteristic is not already disabled -> update it
-      if (ruleDto.getSubCharacteristicId() == null || !RuleDto.DISABLED_CHARACTERISTIC_ID.equals(ruleDto.getSubCharacteristicId())) {
+      if (ruleDto.getSubCharacteristicId() == null || RuleDto.DISABLED_CHARACTERISTIC_ID != ruleDto.getSubCharacteristicId()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)